arch/xtensa/core/vector_handlers.c

```diff
@@ -229,6 +229,21 @@ void *xtensa_excint1_c(int *interrupted_stack)
 	__asm__ volatile("rsr.depc %0" : "=r"(depc));
 
 	is_dblexc = (depc != 0U);
+
+	if (is_dblexc && (cause == EXCCAUSE_DTLB_MISS)) {
```

Comment:
OK, I'm starting to get this.  But walk me through why you need this if() expression?  The cause is in the ZSR register always, no matter whether there was a double exception or not.  We can just fetch it unconditionally and avoid the extra instructions (including the broken one to fetch the essentially useless value above!).  Might be need for a little wrapper API so that MMU=n builds can skip the extra ZSR.